tensorflow_io/audio/kernels/audio_dataset_ops.cc

```diff
@@ -0,0 +1,304 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/core/framework/dataset.h"
+#include "tensorflow/core/lib/io/buffered_inputstream.h"
+#include "tensorflow/core/platform/file_system.h"
+
+namespace tensorflow {
+namespace data {
+namespace {
+
+static const int64 kWAVBufferSize = 256 * 1024;
+
+struct WAVHeader {
+    char riff[4]; // "RIFF"
+    int32 size; // Size after (file size - 8)
+    char wave[4]; // "WAVE"
+    char fmt[4]; // "fmt "
+    int32 fmt_size; // 16 for PCM
+    int16 fmt_type; // 1 for PCM. 3 for IEEE Float
```

Comment:
I don't see `3 for IEEE Float` being supported here though. See my other comment.